### DISCUSSION TITLE
[Question] Sound FX modding

### DISCUSSION THREAD

**Discussion by SlightPersimmon1** [exult/exult#349] [General]
Hello all

I'm getting into sound fx modding, with the primary goal to modernize the sound effects of the game. I figured out how to extract the flx files, how to change the wav's and how to repack them again into the flx. Currently i have a working flx that is successfully working ingame. I have one question though.

Is there a Ultima VII spreadsheet where there is a mapping between the file indexes and a simple description of what they do?
Something like this:

...
00014 - ui open
00024 - big bell
00030 - door open
00031 - door close
....

The reason is that i'm struggling to find some sounds. For instance bird sounds. None of the two existing sound packs (jmsfx.flx and sqsfxbg.flx) seem to be complete, so i have no way to find them out.        
 
Thank you in advance,

Kind regards. 

**[EDIT] I found the following on the old forum. 
http://exult.info/forum/viewtopic.php?f=1&t=1765883&p=1766321&hilit=sound#p1766321**

I'll take a look and see if i find the sounds I'm searching for.

**Comment by DominusExult** [comment]
Some very few are in the soundtrack

**Reply by SlightPersimmon1** [reply]
Thank you for the tip, DominusExult. You are right. It looks like 04bg.ogg to 08bg.ogg also have ambient sounds.

**Reply by DominusExult** [reply]
I have made twopacks ages ago https://www.dropbox.com/s/0qmzb26w2wi7bvb/drbgsfx.flx?dl=0 and https://www.dropbox.com/s/g22thrdb2z2dyip/drsisfx.flx?dl=0
It's a mixture of original sfx, some I lifted elsewhere (I think UO) and some I made myself (ignite was recorded lighting a match).
But since some are lifted from other sources, it's not quite right to advertise much.

The ambient sfx are a problem as they only get played with the digital music option.
It might be worthwhile to make the calls for those music tracks into sfx tracks but that would require code changes. But something worth a feature request perhaps.

**Reply by Zhelkas** [reply]
> I have made twopacks ages ago https://www.dropbox.com/s/0qmzb26w2wi7bvb/drbgsfx.flx?dl=0 and https://www.dropbox.com/s/g22thrdb2z2dyip/drsisfx.flx?dl=0 It's a mixture of original sfx, some I lifted elsewhere (I think UO) and some I made myself (ignite was recorded lighting a match). But since some are lifted from other sources, it's not quite right to advertise much.
> 
> The ambient sfx are a problem as they only get played with the digital music option. It might be worthwhile to make the calls for those music tracks into sfx tracks but that would require code changes. But something worth a feature request perhaps.

Very cool!  How would I get these working in U7 and SI?

**Reply by DominusExult** [reply]
By editing the exult.cfg http://exult.info/docs.php#wave_sfx_install

**Reply by Zhelkas** [reply]
Thanks - I tried that with Black Gate.  I put your files in the Exult Data folder, changed the path in "< waves >" to drbgsfx.flx and nothing seems to have changed.  The sound effects are exactly the same.  Was that the right thing to do?  And what option(s) should be selected in the in-game menu for it to work?

**Comment by DominusExult** [comment]
hmm, if it's done correctly the audio settings should  show "custom". Easiest difference to hear is the opening/closing of doors. 
![Untitled 2](https://github.com/exult/exult/assets/3313896/6ce4aef9-4b58-4737-b93c-c6d51c037d6b)

**Reply by Zhelkas** [reply]
The SFX button shows "Digital", but there's no "Custom" pack option I can find.  It only shows Sound Blaster and Roland.  I also don't have the "CoreAudio" option for the Midi Driver.

**Reply by DominusExult** [reply]
then the cfg is likely having a problem. 
1st make sure you are editing the correct exult.cfg http://exult.info/docs.php#exult_config_location
2nd if you edited the correct one, try changing the <waves> tag for the <forgeofvirtue> entry as well and if that doesn't help please copy/paste your exult.cfg here (but set it in code tags)

Don't worry about Coreaudio, that is macOS specific.

**Reply by Zhelkas** [reply]
That worked!  I was editing the wrong exult.cfg file.  Thanks for your help!  The sounds are great, I recognize a few of them from UO.
